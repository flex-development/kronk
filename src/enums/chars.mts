/**
 * @file Enums - chars
 * @module kronk/enums/chars
 */

import * as vt from '@flex-development/vfile-tokenizer'

/**
 * Character dictionary.
 *
 * @internal
 *
 * @enum {string | null}
 */
const chars = { ...vt.chars, delimiter: vt.chars.hyphen.repeat(2) } as const

export default chars
