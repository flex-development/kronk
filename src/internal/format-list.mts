/**
 * @file Internal - formatList
 * @module kronk/internal/formatList
 */

import chars from '#enums/chars'
import type { List } from '@flex-development/kronk'

/**
 * Create a list string in the form `'A and B'` or `'A, B, ..., and Z`.
 *
 * @internal
 *
 * @param {List<string>} list
 *  The list to format
 * @return {string}
 *  Formatted list
 */
function formatList(this: void, list: List<string>): string {
  list = [...list]

  /**
   * List transition type to insert before last element.
   *
   * @const {string} type
   */
  const type: string = 'and'

  switch (list.length) {
    case 0:
      return chars.empty
    case 1:
      return `${list[0]}`
    case 2:
      return `${list[0]} ${type} ${list[1]}`
    case 3:
      return `${list[0]}, ${list[1]}, ${type} ${list[2]}`
    default:
      return `${list.slice(0, -1).join(', ')}, ${type} ${list.at(-1)}`
  }
}

export default formatList
