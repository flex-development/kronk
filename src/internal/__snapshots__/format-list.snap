// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:internal/formatList > should return list string (0) 1`] = `""`;

exports[`unit:internal/formatList > should return list string (1) 1`] = `"debug"`;

exports[`unit:internal/formatList > should return list string (2) 1`] = `"commonjs and module"`;

exports[`unit:internal/formatList > should return list string (3) 1`] = `"cjs, esm, and iife"`;

exports[`unit:internal/formatList > should return list string (4) 1`] = `"0, 1, 2, and 3"`;

exports[`unit:internal/formatList > should return list string (5) 1`] = `"0, 1, 'dts', false, and true"`;
